kernel/memory_structs/src/lib.rs

```diff
@@ -392,17 +499,12 @@ macro_rules! implement_page_frame_range {
                         
             #[doc = "A range of [`" $chunk "`]s that are contiguous in " $desc " memory."]
             #[derive(Clone, PartialEq, Eq)]
-            pub struct $TypeName(RangeInclusive<$chunk>);
+            pub struct $TypeName<P: PageSize = Page4KiB>(RangeInclusive<$chunk::<P>>);
 
             impl $TypeName {
-                #[doc = "Creates a new range of [`" $chunk "`]s that spans from `start` to `end`, both inclusive bounds."]
```

Comment:
Any reason why you reordered the functions in this impl block? Can you restore the ordering so the diff is more readable?